core/java/com/google/instrumentation/stats/Aggregation.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.Duration;
+import com.google.instrumentation.common.Function;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * An {@link Aggregation} is a series of individual measurements.
+ *
+ * <p> Each measurments can be one of either:
+ * <ul>
+ * <li> {@link DistributionAggregation}
+ * <li> {@link IntervalAggregation}
+ * </ul>
+ */
+public abstract class Aggregation {
+  /**
+   * {@link Tag}s associated with this {@link Aggregation}.
+   *
+   * <p>Note: The returned list is unmodifiable, attempts to update it will throw an
+   * UnsupportedOperationException.
+   */
+  public final List<Tag> getTags() {
+    return tags;
+  }
+
+  /**
+   * Applies the given match function to the underlying data type.
+   */
+  public abstract <T> T match(
+      Function<DistributionAggregation, T> p0,
+      Function<IntervalAggregation, T> p1);
+
+  private final List<Tag> tags;
+
+  // Seals the tagged union.
+  private Aggregation(List<Tag> tags) {
```

Comment:
The comment is a little unclear.  Should it say something like "constructor is private in order to seal the tagged union"?